### PR TITLE
Combat Robots no longer have organs

### LINKED ISSUE #9256: Robots get heart damage and shrapnel
### Description

Read title

### Test Merges

none

### Reproduction Steps

dunno

### Screenshots

image

### PR DESCRIPTION
**Pull request by Three333Digit**
## About The Pull Request
Title. Closes #9256
Combat robots no longer have organs.
~~The shrapnel change also affects anything robotic, that means synths and any human with robotic limbs gained through surgery won't get shrapnel in that specific robotic limb. If that's not good, tell me.~~

## Why It's Good For The Game
Bugfix. ~~The main reason why combat robots get heart damage is defibs from what I've found.~~ Combat robots no longer get organ damage... because they no longer have organs.

## Changelog
:cl:
del: combat robots no longer have organs
/:cl:

### CODE REVIEW

**Comment by Lumipharon** [comment]
What's the big deal with shrapnel? Just being a beepboop doesn't mean bullets can't lodge themselves in you, and they can still be removed by tweezer afaik.

**Comment by jroinc** [comment]
If you have a chance, I'd go into organs/organs_internal.dm OR carbon/human/species.dm and add, say, a 1 pt/tick autoheal for either prosthetic organs OR robot organs, given as we're assuming that they shouldn't be hurt by anything. That'll catch any edge cases that slip through.

**Comment by ilikesaying** [comment]
> What's the big deal with shrapnel? Just being a beepboop doesn't mean bullets can't lodge themselves in you, and they can still be removed by tweezer afaik.

shrapnel damage is worse to heal since robots can't take medicine to automatically heal the damage, meaning they get damage every time they walk that they have to manually repair while searching for the closest corpsman

**Comment by Moggram** [comment]
could have it so that wirecutters remove shrapnel.

**Comment by Lumipharon** [comment]
> > What's the big deal with shrapnel? Just being a beepboop doesn't mean bullets can't lodge themselves in you, and they can still be removed by tweezer afaik.
> 
> shrapnel damage is worse to heal since robots can't take medicine to automatically heal the damage, meaning they get damage every time they walk that they have to manually repair while searching for the closest corpsman

True, but robots are second class citizens, if they choose to play robots they should accept the drawbacks. Not being able to get shrapnel at all is a straight upgrade compared to meatbags.

**Comment by Naaanii** [comment]
I'd say keep shrap but remove organ damage if possible.

**Comment by Mrrpip** [comment]
> > > What's the big deal with shrapnel? Just being a beepboop doesn't mean bullets can't lodge themselves in you, and they can still be removed by tweezer afaik.
> > 
> > 
> > shrapnel damage is worse to heal since robots can't take medicine to automatically heal the damage, meaning they get damage every time they walk that they have to manually repair while searching for the closest corpsman
> 
> True, but robots are second class citizens, if they choose to play robots they should accept the drawbacks. Not being able to get shrapnel at all is a straight upgrade compared to meatbags.

But we have ballistic armor?

**Comment by Losenis** [comment]
> True, but robots are second class citizens, if they choose to play robots they should accept the drawbacks. Not being able to get shrapnel at all is a straight upgrade compared to meatbags.

Pretty sure the lack of boots and gloves, a new loadout altogether, the inability of using chems, being a robot, effectively being critdrag bait because no oxyloss on crit, being conveniently FFd when you earrape someone with warcry, and inability to run are wild enough drawbacks.

**Review by BraveMole — CHANGES_REQUESTED** [review]
the heart damage is a straight bug fix, cool.

The shrap one, eh, i'm not a fan. 
Removing organ damage (or organs) would be cool though

**Comment by Three333Digit** [comment]
Might need to TM first.

File: code/game/objects/items/scanners.dm [hunk 1]
```diff
@@ -190,7 +190,7 @@ REAGENT SCANNER
 	// Show red messages - broken bokes, infection, etc
 	if (M.getCloneLoss())
 		dat += "\t[span_scanner(" *Cellular damage detected. Recommend cryogenic treatment")].\n"
-	if (M.getBrainLoss() >= 100 || !M.has_brain())
+	if (M.getBrainLoss() >= 100 || !M.has_brain() && !isrobot(M))
```
Comment (BraveMole) [inline code/game/objects/items/scanners.dm:193]: Logic seems flawed here

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Review by BraveMole — APPROVED** [review]

**Pull request merged**